File: backend/app/schemas/artifact.py
import uuid
from datetime import datetime

from pydantic import BaseModel, Field

from app.models.enums import ArtifactType, ArtifactChannel, ArtifactFormat, ArtifactStatus


class ArtifactCreatorResponse(BaseModel):
    id: uuid.UUID
    name: str
    avatar_url: str | None = None

    model_config = {"from_attributes": True}


class ArtifactResponse(BaseModel):
    id: uuid.UUID
    project_id: uuid.UUID
    creator: ArtifactCreatorResponse
    type: ArtifactType
    name: str
    channel: ArtifactChannel | None = None
    format: ArtifactFormat | None = None
    thumbnail_url: str | None = None
    compliance_score: float | None = None
    status: ArtifactStatus
    version: int
    created_at: datetime

    model_config = {"from_attributes": True}


class ArtifactDetailResponse(ArtifactResponse):
    content: dict | None = None
    locks: list[str] | None = None
    video_session_id: uuid.UUID | None = None


class ArtifactListResponse(BaseModel):
    items: list[ArtifactResponse]
    total: int
    page: int
    per_page: int


class CreateArtifactRequest(BaseModel):
    type: ArtifactType
    name: str = Field(min_length=1, max_length=255)
    content: dict | None = None
    channel: ArtifactChannel | None = None
    format: ArtifactFormat | None = None
    target_duration_seconds: int | None = None  # VIDEO/REEL only; defaults to 60


class UpdateArtifactRequest(BaseModel):
    name: str | None = Field(None, min_length=1, max_length=255)
    content: dict | None = None
    channel: ArtifactChannel | None = None
    format: ArtifactFormat | None = None

File: frontend/src/types/artifact.ts
export type ArtifactType = "poster" | "whatsapp_card" | "reel" | "story" | "video" | "deck" | "infographic" | "slide_deck";
export type ArtifactChannel = "instagram" | "whatsapp" | "print" | "social" | "internal";
export type ArtifactFormat = "1:1" | "4:5" | "9:16" | "A4" | "800x800";
export type ArtifactStatus = "draft" | "ready" | "exported";

export interface ArtifactCreator {
  id: string;
  name: string;
  avatar_url: string | null;
}

export interface Artifact {
  id: string;
  project_id: string;
  creator: ArtifactCreator;
  type: ArtifactType;
  name: string;
  channel: ArtifactChannel | null;
  format: ArtifactFormat | null;
  thumbnail_url: string | null;
  compliance_score: number | null;
  status: ArtifactStatus;
  version: number;
  created_at: string;
}

export interface ArtifactDetail extends Artifact {
  content: Record<string, unknown> | null;
  locks: string[] | null;
  video_session_id: string | null;
}

export interface ArtifactListResponse {
  items: Artifact[];
  total: number;
  page: number;
  per_page: number;
}
